### PR TITLE
[MISC] move seqan3::views::to to seqan3/utility/views/

### PR DESCRIPTION
**Pull request by MitraDarja**
Resolves https://github.com/seqan/product_backlog/issues/355

### CODE REVIEW

File: include/seqan3/utility/views/to.hpp [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+namespace seqan3::views
+{
+
+/*!\brief A to view
+ * \ingroup views
+ * \details
+ * \noapi{This is currently range-v3's to implementation.}
+ */
+inline constexpr auto to = ::ranges::views::to;
```
Comment (marehr) [inline include/seqan3/utility/views/to.hpp:28]:
```suggestion
inline constexpr auto to = ::ranges::to;
```

File: include/seqan3/utility/views/to.hpp [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+namespace seqan3::views
+{
+
+/*!\brief A to view
+ * \ingroup views
+ * \details
+ * \noapi{This is currently range-v3's to implementation.}
+ */
+inline constexpr auto to = ::ranges::to;
```
Comment (marehr) [inline include/seqan3/utility/views/to.hpp:28]:
```suggestion
#if SEQAN3_DOXYGEN_ONLY(1)0
inline constexpr auto to;
#else // ^^^ doxygen only / real import vvv
using ::ranges::to;
#endif // SEQAN3_DOXYGEN_ONLY(1)0
```

File: include/seqan3/utility/views/to.hpp [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ * \author Hannes Hauswedell <hannes.hauswedell AT fu-berlin.de>
+ * \brief Provides seqan3::views::to.
+ */
+
+#pragma once
+
+#include <range/v3/range/conversion.hpp>
+
+#include <seqan3/core/platform.hpp>
+#include <seqan3/std/ranges>
```
Comment (marehr) [inline include/seqan3/utility/views/to.hpp:18]:
```suggestion
#include <seqan3/std/ranges>

#include <range/v3/range/conversion.hpp>

#include <seqan3/core/platform.hpp>
```

File: include/seqan3/io/sam_file/detail/format_sam_base.hpp [hunk 4]
```diff
@@ -39,12 +39,12 @@
 #include <seqan3/io/sequence_file/output_options.hpp>
 #include <seqan3/range/detail/misc.hpp>
 #include <seqan3/range/views/take_until.hpp>
-#include <seqan3/range/views/to.hpp>
 #include <seqan3/utility/detail/exposition_only_concept.hpp>
 #include <seqan3/utility/detail/type_name_as_string.hpp>
 #include <seqan3/utility/tuple/concept.hpp>
 #include <seqan3/utility/views/repeat_n.hpp>
 #include <seqan3/utility/views/slice.hpp>
+#include <seqan3/utility/views/to.hpp>
```
Comment (Irallia) [inline include/seqan3/io/sam_file/detail/format_sam_base.hpp:47]: Is this include needed, or can it be removed?

File: include/seqan3/search/fm_index/fm_index.hpp [hunk 5]
```diff
@@ -21,10 +21,10 @@
 #include <seqan3/alphabet/views/to_rank.hpp>
 #include <seqan3/core/range/type_traits.hpp>
 #include <seqan3/range/views/join.hpp>
-#include <seqan3/range/views/to.hpp>
 #include <seqan3/search/fm_index/concept.hpp>
 #include <seqan3/search/fm_index/detail/fm_index_cursor.hpp>
 #include <seqan3/search/fm_index/fm_index_cursor.hpp>
+#include <seqan3/utility/views/to.hpp>
```
Comment (Irallia) [inline include/seqan3/search/fm_index/fm_index.hpp:27]: Is this include needed, or can it be removed?

File: test/performance/alignment/global_affine_alignment_simd_benchmark_template.hpp [hunk 6]
```diff
@@ -15,10 +15,10 @@
 #include <seqan3/alignment/pairwise/align_pairwise.hpp>
 #include <seqan3/alphabet/aminoacid/aa20.hpp>
 #include <seqan3/alphabet/nucleotide/dna4.hpp>
-#include <seqan3/range/views/to.hpp>
 #include <seqan3/test/performance/sequence_generator.hpp>
 #include <seqan3/test/performance/units.hpp>
 #include <seqan3/test/seqan2.hpp>
+#include <seqan3/utility/views/to.hpp>
```
Comment (Irallia) [inline test/performance/alignment/global_affine_alignment_simd_benchmark_template.hpp:21]: Is this include needed, or can it be removed?

**Review by Irallia — APPROVED** [review]
LGFM, just some includes maybe unsused.

File: include/seqan3/range/views/to.hpp [hunk 7]
```diff
@@ -7,21 +7,13 @@
...
-
-namespace seqan3::views
-{
-
-//TODO reimplement me
-using ::ranges::to;
-
-} // namespace seqan3::views
+SEQAN3_DEPRECATED_HEADER(
+    "This header is deprecated and will be removed in SeqAn-3.1.0; Please #include <seqan3/utility/views/to.hpp> instead.")
```
Comment (Irallia) [inline include/seqan3/range/views/to.hpp:19]: I don't know if I didn't see it before, or if it came from the changes: newline is missing here.

**Comment by Irallia** [comment]
After I asked you to delete some includes, I guess one is missing in: `seqan3/test/performance/search/index_construction_benchmark.cpp`

**Review by marehr — APPROVED** [review]

**Pull request merged**